src/controllers/red-experience.js

```diff
@@ -0,0 +1,74 @@
+import {ReturnState} from './_base.js';
+
+/**
+ * Clean the incoming POST request body to make it more compatible with the
+ * database and its validation rules.
+ *
+ * @param {object} body The incoming request's body.
+ * @returns {any} A json object that's just got our cleaned up fields on it.
+ */
+const cleanInput = (body) => {
+  return {
+    // Ensure the user has entered integers
+    cleanRedExperienceYears: body.redExperienceYears === undefined ? undefined : body.redExperienceYears.trim(),
+    cleanRedControlNumber: body.redControlNumber === undefined ? undefined : body.redControlNumber.trim()
+  };
+};
+
+const redExperienceController = (request) => {
+  // Clear any existing errors.
+  request.session.experienceError = false;
+  request.session.redSelectionError = false;
+  request.session.redEmptyError = false;
+  request.session.redExperienceYearsInvalidError = false;
+  request.session.redControlNumberInvalidError = false;
+
+  if (request.body.red === '' || request.body.red === undefined) {
+    request.session.redSelectionError = true;
+  }
+
+  if (request.body.red === 'yes') {
+    // Clean up the user's input before we use it.
+    const {cleanRedExperienceYears, cleanRedControlNumber} = cleanInput(request.body);
+    request.session.red = 'yes';
+    request.session.redExperienceYears = cleanRedExperienceYears;
+    request.session.redControlNumber = cleanRedControlNumber;
+
+    if (request.session.redExperienceYears) {
+      request.session.redExperienceYearsInvalidError = new RegExp(/\D/).test(cleanRedExperienceYears);
+    }
+
+    if (request.session.redControlNumber) {
+      request.session.redControlNumberInvalidError = new RegExp(/\D/).test(cleanRedControlNumber);
+    }
```

Comment:
If you do end up using the util/cleanInt function, these tests won't work correctly any more and should be done on the trimmed, user supplied strings.